### PR TITLE
feat(STONEINTG-371): Don't add a component to Snapshots if it doesn't have a container image defined

### PR DESCRIPTION
**Pull request by 14rcole**
The acceptance criteria stated "Updates for relevant unit/controller tests".  I believe I covered all relevant tests.  Please let me know if there are other places where I should be adding additional checks.

### CODE REVIEW

File: controllers/pipeline/pipeline_adapter.go [hunk 1]
```diff
@@ -412,6 +412,14 @@ func (a *Adapter) prepareSnapshot(application *applicationapiv1alpha1.Applicatio
 			// Get ComponentSource for the component which is not built in this pipeline
 			componentSource = a.getComponentSourceFromComponent(&applicationComponent)
 		}
+
+		// If containerImage is empty, we have run into a race condition in
+		// which multiple components are being built in close succession.
+		// We omit this not-yet-built component from the snapshot rather than
+		// including a component that is incomplete.
+		if containerImage == "" {
```
Comment (Josh-Everett) [inline controllers/pipeline/pipeline_adapter.go:420]: Do we need to be concerned by the possibility of container Image being nil?

**Comment by Josh-Everett** [comment]
Looks good, just curious about nil condition. Think this is ready to open up to others as well.

**Comment by 14rcole** [comment]
I think the failure in the unit tests is another race condition.  The test in question, `It ensures the global component list is changed and compositeSnapshot should be created`, creates a composite snapshot, waits for the snapshot CR to exist with an `Eventually` block, then attempts to create the snapshot again to ensure that the function correctly identifies the existing snapshot CR and returns it rather than attempting to create a new one.  Most of the time when I run `make test` locally I run into the same error.  Occassionally the error doesn't occur.  I think the best option is to increase the timeout of this eventually block but I'm going to verify during standup today that there's not a better option.

**Comment by dirgim** [comment]
The [gitlint check](https://github.com/redhat-appstudio/integration-service/actions/runs/4689564616/jobs/8314106779?pr=146) is failing because of the title of the commit needing to be compatible with the conventional commit format, so it would need to be changed to something like this: `feat(STONEINTG-371): snapshot components must have container images`

**Comment by dirgim** [comment]
LGTM, this just needs a rebase on the latest master to fix e2e tests

**Pull request merged**